app/controllers/ajax_controller/primers.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# see ajax_controller.rb
+class AjaxController
+  # Get list of names for autocompletion in mobile app.
+  def name_primer
+    render(json: name_list)
+  end
+
+  # Get list of locations for autocompletion in mobile app.
+  def location_primer
+    render(json: location_list)
+  end
+
+  private
+
+  def name_list
+    name_ids = Observation.select(:name_id).distinct
+    fields = [:id, :text_name, :author, :deprecated, :synonym_id]
+    names = Name.where(id: name_ids).select(*fields)
```

Comment:
If you feel like it, you could eliminate one call to the db (and one assignment:
```suggestion
    fields = [:id, :text_name, :author, :deprecated, :synonym_id]
    names = Name.where(id: Observation.select(:name_id).distinct).select(*fields)
```